vault/logical_system.go

```diff
@@ -1644,9 +1644,13 @@ func (b *SystemBackend) handleMount(ctx context.Context, req *logical.Request, d
 	// is enabled in dev and test servers. We don't want to pay the cost of key
 	// generation for that KV mount in all tests.
 	if config.usingOIDCDefaultKey() && logicalType != mountTypeKV {
-		if err := identityStore.lazyGenerateDefaultKey(ctx, storage); err != nil {
+		err := identityStore.lazyGenerateDefaultKey(ctx, storage)
+		if err != nil && !errors.Is(err, logical.ErrReadOnly) {
 			return nil, fmt.Errorf("failed to generate default key: %w", err)
 		}
+
+		b.Logger().Warn("skipping default OIDC key generation for local mount",
```

Comment:
How do we know this is a local mount?

Reply:
It being a local mount is the only context I can think of where the mount request would not be forwarded. For example, forwarding happens here https://github.com/hashicorp/vault/blob/main/vault/logical_system.go#L3107-L3112. I also assume performance standbys are forwarded on the first read-only storage error they see.

I have information on whether it's a local mount in the handler. I will condition the log on that.

Reply:
I've updated this to only log a warning if we know we're a local mount in https://github.com/hashicorp/vault/pull/25265/commits/c41b75af6bff8c0c08abe21e74d0b0e8026af14e.